### PR TITLE
Add Calamares: A distribution independent installer framework.

### PR DESCRIPTION
**Pull request by ts468**
The PR adapts boost lib as well, so it probably expensive to merge.

### CODE REVIEW

**Comment by ts468** [comment]
The idea is to maybe build a graphical installer for NixOS with calamares.

**Comment by ts468** [comment]
The PR builds locally for me, and the changes to boost lib should be fairly easy to verify.

File: pkgs/development/libraries/boost/generic.nix [hunk 1]
```diff
@@ -149,7 +149,7 @@ stdenv.mkDerivation {
   configureScript = "./bootstrap.sh";
   configureFlags = commonConfigureFlags ++ [
     "--with-icu=${icu}"
-    "--with-python=${python}/bin/python"
+    "--with-python=${python}/bin/python${python.majorVersion}"
```
Comment (copumpkin) [inline pkgs/development/libraries/boost/generic.nix]: Why is this needed? I see you mention that you change boost, but not why

Comment (ts468) [inline pkgs/development/libraries/boost/generic.nix]: boost can be build with python2 and python3. The python2 package has a symlink to python and so works with the previous configuration, however python3 does not have the symlink and so doesn't work.

Comment (copumpkin) [inline pkgs/development/libraries/boost/generic.nix]: Could the python package just be amended to have a `python -> python3` symlink? Or is that unconventional? It just doesn't feel like boost should need to be concerned about python version information like that.

Comment (ts468) [inline pkgs/development/libraries/boost/generic.nix]: Well, if a symlink would be added, then python2 and python3 would probably collide if installed together. I assume that was the original reason to not create a symlink for python3?

Comment (copumpkin) [inline pkgs/development/libraries/boost/generic.nix]: I suppose that makes some sense. Eyeballing the python expressions, it looks like they all have an `interpreter` pass-through attribute which is set to suitable values. Could you just use `--with-python=${python.interpreter}`, perhaps? It would still cause a boost rebuild, but would feel less "implementation detailey".

**Comment by ts468** [comment]
@copumpkin, thanks for the review! I've updated boost lib accordingly.

**Comment by ts468** [comment]
@copumpkin, could you maybe have another look at that PR, please?

**Comment by ts468** [comment]
@copumpkin, could you maybe have a look at this PR again, please? Or, should I ask someone else to merge the PR? I just don't want the PR to be forgotten. Many thanks!

**Comment by thoughtpolice** [comment]
I assume if updating Boost will cause an expensive rebuild, we should put it into `staging`.

**Comment by 7c6f434c** [comment]
Given that manu boost updates break some packages, adding a new version in master as non-default is a reasonable option.


-------- Original Message --------

**Comment by ts468** [comment]
@thoughtpolice, does that mean I have to open a PR against the staging branch?

@7c6f434c, I don't expect the change to break any packages. Either, they use python2 so far, in which case the change is transparent, or they use python3, in which case boost doesn't compile so far.
